packages/modules/order/src/utils/actions/change-shipping-address.ts

```diff
@@ -0,0 +1,20 @@
+import { ChangeActionType } from "@medusajs/framework/utils"
+
+import { OrderChangeProcessing } from "../calculate-order-change"
+import { setActionReference } from "../set-action-reference"
+
+OrderChangeProcessing.registerActionType(
+  ChangeActionType.CHANGE_SHIPPING_ADDRESS,
```

Comment:
thought: I would be curious to hear @carlos-r-l-rodrigues's thoughts on creating a more generic action, that can be used for updates that does not have any validation and where the operation is basically just updating the property on the order.

For example:
```ts
OrderChangeProcessing.registerActionType(
  ChangeActionType.ORDER_UPDATE,
  {
    operation({ action, currentOrder, options }) {
      /**
       * NOOP: used as a reference for the change
       */

      setActionReference(currentOrder, action, options)
    },
    validate({ action }) {
      /* noop */
    },
  }
)
```

The order change action generated for this type of update would then be:
```ts
const actionInput = transform(
   { order, input, change },
   ({ order, input, change }) => [
     {
       order_change_id: change.id,
       order_id: input.order_id,
       action: ChangeActionType.ORDER_UPDATE,
       version: change.version,
       reference: "shipping_address", //  <-------------- this would be dynamic
       applied: true,
       reference_id: order.shipping_address?.id, //  <-------------- this would be dynamic
       details: input.shipping_address as Record<string, unknown>, //  <-------------- this would be dynamic
     },
   ]
 )
```

This can be used for `email`, `shipping_address`, and `billing_address`. 

Perhaps, a question to ask: Do we gain anything from having the specific actions? If not, we made as well save ourselves from having multiple actions for these updates.

Reply:
Currently, that action accepts anything, even if there is no "action" associated with it on the Order module.
So, it is really up to us to decide. To me, and ORDER_UPDATE seems to fit well, and anything to distinguish them on the UI, can go on the details